### PR TITLE
Set status explicitly in the kube tag

### LINKED ISSUE #4: handle status
Status needs a [`#[kube(status = ...)]`](https://docs.rs/kube/0.64.0/kube/derive.CustomResource.html#kubestatus--statusstructname) instruction.

Should be present in crds whose yaml contains:

```
    subresources:
      status: {}
```

which should be detectable through the [CustomResourceDefinition's picked Version](https://arnavion.github.io/k8s-openapi/v0.13.x/k8s_openapi/apiextensions_apiserver/pkg/apis/apiextensions/v1/struct.CustomResourceDefinitionVersion.html)

### PR DESCRIPTION
**Pull request by flaper87**
Without this, the status data won't be accessible through the generated struct

fixes #4

Signed-off-by: Flavio Percoco <flaper87@gmail.com>

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -107,8 +107,8 @@ impl Kopium {
                     if s.level == 1 && s.name.ends_with("Spec") {
                         self.print_derives();
                         println!(
-                            r#"#[kube(group = "{}", version = "{}", kind = "{}", plural = "{}")]"#,
-                            group, version, kind, plural
+                            r#"#[kube(group = "{}", version = "{}", kind = "{}", plural = "{}", status = "{}Status")]"#,
```
Comment (clux) [inline src/main.rs:110]: this is right, but it needs to be done conditionally. it needs to check in the crd type if the status subresource is set first

Comment (flaper87) [reply src/main.rs]: ah, good point! Pushing a new version

File: src/main.rs [hunk 2]
```diff
@@ -113,6 +113,10 @@ impl Kopium {
                         if scope == "Namespaced" {
                             println!(r#"#[kube(namespaced)]"#);
                         }
+
+                        if crd.status.is_some() {
```
Comment (clux) [inline src/main.rs:117]: this needs to be something like:

```suggestion
                        if let Some(subres) = version.subresources.status {
                            if subres.status.is_some() {
```

Comment (clux) [reply src/main.rs:117]: the info comes from the version struct: https://arnavion.github.io/k8s-openapi/v0.14.x/k8s_openapi/apiextensions_apiserver/pkg/apis/apiextensions/v1/struct.CustomResourceDefinitionVersion.html

Comment (flaper87) [reply src/main.rs]: Changed a bit your suggestion, thanks for the pointer (quite new to the various libraries here)

**Comment by flaper87** [comment]
@clux let's get #33 in first and then we can test this one on top of that one :D

**Comment by clux** [comment]
Yeah, good plan. We probably need to do some changes afterwards as well to get the status object patched in (kubectl doesn't apply statuses - so currently no way to verify the instance you sent)

**Comment by clux** [comment]
main is updated if you want to update

**Comment by flaper87** [comment]
> main is updated if you want to update

done :)

File: src/main.rs [hunk 3]
```diff
@@ -159,11 +159,16 @@ impl Kopium {
...
                         println!(
                             r#"#[kube(group = "{}", version = "{}", kind = "{}", plural = "{}")]"#,
-                            group, version, kind, plural
+                            group, version_name, kind, plural
                         );
                         if scope == "Namespaced" {
                             println!(r#"#[kube(namespaced)]"#);
                         }
+
+                        if let Some(CustomResourceSubresources { status: Some(_), .. }) = version.subresources
```
Comment (clux) [inline src/main.rs:168]: ah that is a nice pattern match :+1:

**Review by clux — APPROVED** [review]

**Comment by clux** [comment]
thanks a lot - tests are passing :-)

**Pull request merged**